tests/improver-generate-topographybands-ancillary/03-bounds_json.bats

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bats
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+. $IMPROVER_DIR/tests/lib/utils
+
+@test "generate-topographybands-ancillary input_orog.nc input_land.nc output.nc" {
+  TEST_DIR=$(mktemp -d)
+  improver_check_skip_acceptance
+  test_path=$IMPROVER_ACC_TEST_DIR/generate-topographybands/basic/
+
+  # Run topography band ancillary generation and check it passes.
+  run improver generate-topographybands-ancillary \
+      "$test_path/input_orog.nc" \
+      "$test_path/input_land.nc" \
+      "$TEST_DIR/output.nc" \
+      --thresholds $test_path/bounds.json
```

Comment:
shouldn't this be --thresholds_filepath

When I ran the tests the use of --thresholds did not break the test.

Reply:
I assume that it doesn't break because the --thresholds option is ignored. It might be best to choose different bounds for the default dictionary and this CLI test, so that it's more obvious whether the default thresholds or the thresholds provided in the json file are being used.

Reply:
I initially thought that --thresholds is just ignored but if I add something that isn't in the CLI e.g. --bob it falls over. Could argsparse be matching --thresholds to --thresholds_filepath as it is a partial match?

Reply:
Yes, I think that you're right! This is described in https://docs.python.org/2.7/library/argparse.html#argument-abbreviations-prefix-matching, although I wouldn't have anticipated that this would be the default behaviour. According to https://stackoverflow.com/questions/33900846/disable-unique-prefix-matches-for-argparse-and-optparse, there is an option in Python 3.5 and greater to disable this feature: https://docs.python.org/3.6/library/argparse.html#allow-abbrev.

Reply:
I considered this, and will be happy to change the json to make sure that this is ok.

Reply:
That's almost too clever for comfort. I'm not sure I like library writers trying to second guess me like that.